Chore/type graphql

Sam09040 opened the pull request:
Pull request focusing only on adding TypeGraphql

Sam09040 committed:
Chore/type-graphql

- Added TypeGraphql, class-validator and reflect-metada;
- Added types and inputs to module;
- Changed from using `ApolloServer` to `startStandaloneServer`;
- Using buildSchema to get resolvers;

racsandoval commented on src/api/graphql/module/address/address.resolver.ts:
```diff
@@ -1,11 +1,19 @@
+export class AddressResolver {
+  constructor() {}
+
+  @Query(() => [Address], { description: 'Get addresses' })
+  async address(@Arg('userId', () => Int) userId: number) {
+    return addressUseCase(userId);
+  }
+
+  @Mutation(() => Address, { description: 'Create a new address' })
+  async createAddress(@Arg('data') data: AddressInput) {
+    return createAddressUseCase(data);
+  }
```
essas chamadas não deveriam utilizar o context para verificar se o usuário está logado? Também poderia utilizar o decorator `Authorized` do `type-graphql` para facilitar algumas verificações

Sam09040 replied:
Durante o botinho não foi pedido que essas operações fossem autorizadas, mas posso adicionar!

racsandoval commented on src/api/graphql/module/user/login.type.ts:
```diff
@@ -0,0 +1,12 @@
+
+@ObjectType({ description: 'Login response object' })
+export class Login implements LoginModel {
+  @Field({ description: 'JWT token' })
+  token: string;
```
o token não deveria ser enviado por input, deveria ser pego pela header da chamada

Sam09040 replied:
essa é a resposta do login, é enviado para o usuário quando o login tem sucesso.

racsandoval replied:
viajei então kkk

racsandoval commented on src/api/graphql/server.context.ts:
```diff
@@ -0,0 +1,8 @@
+}
+
+export const context = async ({ req }: any): Promise<ServerContext> => {
+  const token = req.headers.authorization ?? undefined;
+  return { token };
```
que tal já decodificar o token aqui e retornar os dados do payload?

Sam09040 committed:
fix: context and input

- added authorized decorator;
- using class-validator for validation;
- added userId to context data;
- added validation for input types with class-validator;
